src/components/Modals/PublishCollectionModal/PublishCollectionModal.tsx

```diff
@@ -1,39 +1,120 @@
 import * as React from 'react'
-import { ModalNavigation, Button } from 'decentraland-ui'
+import { ModalNavigation, Button, Mana, Loader } from 'decentraland-ui'
 import Modal from 'decentraland-dapps/dist/containers/Modal'
 import { t } from 'decentraland-dapps/dist/modules/translation/utils'
 
-import { Props } from './PublishCollectionModal.types'
+import { builder } from 'lib/api/builder'
+import { fromWei } from 'web3x-es/utils'
+import { RARITY_COLOR, ItemRarity } from 'modules/item/types'
+import { Props, State } from './PublishCollectionModal.types'
 import './PublishCollectionModal.css'
 
-export default class PublishCollectionModal extends React.PureComponent<Props> {
-  componentDidMount() {
+export default class PublishCollectionModal extends React.PureComponent<Props, State> {
+  state: State = { step: 1, rarities: [], isFetchingRarities: true }
+
+  async componentDidMount() {
     const { collection, onClose } = this.props
     if (!collection) {
       onClose()
     }
+
+    const rarities = await builder.fetchRarities()
+    this.setState({ rarities, isFetchingRarities: false })
   }
 
   handlePublish = () => {
     const { collection, items, onPublish } = this.props
     onPublish(collection!, items)
   }
 
+  handleProceed = () => {
+    this.setState({ step: 2 })
+  }
+
+  renderFirstStep = () => {
+    const { items } = this.props
+    const { rarities, isFetchingRarities } = this.state
+
+    const itemsByRarity: Record<string, { name: ItemRarity; count: number; price: number }> = {}
+    let totalPrice = 0
+
+    if (rarities.length > 0) {
+      items.forEach(item => {
+        const rarity = rarities.find(r => r.name === item.rarity)!
+        if (!itemsByRarity[rarity.name]) {
+          itemsByRarity[rarity.name] = { name: rarity.name, count: 0, price: 0 }
+        }
+
+        const rarityPrice = fromWei(rarity.price, 'ether')
+        itemsByRarity[rarity.name].count++
+        itemsByRarity[rarity.name].price += parseInt(rarityPrice, 10)
+        totalPrice += parseInt(rarityPrice, 10)
+      })
+    }
+
+    return (
+      <>
+        {isFetchingRarities ? (
+          <div className="loader-wrapper">
+            <Loader size="big" active={isFetchingRarities} />
+          </div>
+        ) : (
+          <>
+            {t('publish_collection_modal.items_breakdown_title', { items: items.length })}
+            <div className="items-breakdown">
+              {Object.values(itemsByRarity).map(itemByRarity => {
+                return (
```

Comment:
We can change this to
```suggestion
              {Object.values(itemsByRarity).map(itemByRarity => 
                 (
```